### ISSUE TITLE
Idempotency - not working?

### ISSUE THREAD

**Issue by ortschun** [inngest/inngest#875]
### Discussed in https://github.com/orgs/inngest/discussions/874

<div type='discussions-op-text'>

<sup>Originally posted by **ortschun** November 24, 2023</sup>
Hey folks, I'm new to inngest and loving the product so far. Currently, I am having difficulties with getting the idempotency key to work. I might be missing something regarding the functionality.

The behavior I am expecting is, when there's a send event, if the id field is set, it will only run a single send call for a given particular id and drop the other send events.

I'm on nextjs, and have this function below using the same const idempotencyKey, testing on local development stack, with local inngest running, called literally one after the other.

```
await inngest.send({
    id: idempotencyKey,
    name: CREATE_GOOGLE_CALENDAR_INVITE_EVENT_NAME,
    data: data,
  });
  await inngest.send({
    id: idempotencyKey,
    name: CREATE_GOOGLE_CALENDAR_INVITE_EVENT_NAME,
    data: data,
  });
  await inngest.send({
    id: idempotencyKey,
    name: CREATE_GOOGLE_CALENDAR_INVITE_EVENT_NAME,
    data: data,
  });
```

I expect this to trigger only once but I see all 3 events succeeding, creating 3 calendar invites. Would love to understand if I'm not using idempotency in the intended way, or running into an inngest issue!
</div>

**Comment by halfbakedsneed** [comment]
Yep, we're observing the exact same behaviour, but **_only when using the local dev server_**. In cloud environments, `id` functions as you'd expect.

**Comment by murty0** [comment]
still an issue

In my case, my code was checking that each event has a unique id, and no 2 events with the same id would be triggered. But still, multiple events are triggered for each invocation, and idempotency isnt being respected at all.

happens only in local inngest setup. Workaround is to switch to remote Inngest even for testing.
